plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/DefaultThriftMetastoreClientFactory.java

```diff
@@ -233,4 +303,23 @@ private static void validateCertificates(KeyStore keyStore)
             }
         }
     }
+
+    private record ThriftHttpContext(Optional<String> token, Map<String, String> additionalHeaders)
```

Comment:
ThriftHttpContext is exposed outside of this class because of public method though it's private record.

Reply:
Changed the public method to private as suggested earlier.